--- transformers/modeling_utils.py
+++ transformers/modeling_utils.py
@@ -4023,6 +4023,12 @@
             ):
                 device_map_kwargs["force_hooks"] = True
             if (
+                "force_hooks" in inspect.signature(dispatch_model).parameters
+                and hf_quantizer is not None
+                and hf_quantizer.quantization_config.quant_method == QuantizationMethod.BITS_AND_BYTES
+            ):
+                device_map_kwargs["force_hooks"] = True
+            if (
                 hf_quantizer is not None
                 and hf_quantizer.quantization_config.quant_method == QuantizationMethod.FBGEMM_FP8
                 and isinstance(device_map, dict)
